core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1534,27 +1538,31 @@ public void onReceive(Context context, Intent intent) {
         public void onReceive(Context context, Intent intent) {
             // sound is about to change, eg. bluetooth -> speaker
             Log.d(TAG, "Pausing playback because audio is becoming noisy");
-            pauseIfPauseOnDisconnect();
+            pauseIfPauseOnDisconnect(AudioOutputMode.PHONE_SPEAKER);
         }
-        // android.media.AUDIO_BECOMING_NOISY
     };
 
     /**
      * Pauses playback if PREF_PAUSE_ON_HEADSET_DISCONNECT was set to true.
      */
-    private void pauseIfPauseOnDisconnect() {
-        if (UserPreferences.isPauseOnHeadsetDisconnect() && !isCasting()) {
-            if (mediaPlayer.getPlayerStatus() == PlayerStatus.PLAYING) {
-                transientPause = true;
-            }
-            mediaPlayer.pause(!UserPreferences.isPersistNotify(), true);
+    private void pauseIfPauseOnDisconnect(AudioOutputMode outputMode) {
+        Log.d(TAG, "pauseIfPauseOnDisconnect() " + outputMode);
+        if (UserPreferences.isPauseOnHeadsetDisconnect()
+                && outputMode == AudioOutputMode.PHONE_SPEAKER
+                && !isCasting()) {
+            transientPause = true;
+            mediaPlayer.pause(!UserPreferences.isPersistNotify(), false);
```

Comment:
When headphone is disconnected, and the output is going to the phone speaker, then pause.  This would allow us to continue if the output would go to bluetooth